bot/cogs/models/waifuListApi.py

```diff
@@ -0,0 +1,295 @@
+        """Returns XSRF token."""
+        if self._xsrf_token is None:
+            await self._acquire_tokens_for_forgery()
+        return self._xsrf_token
+
+    @property
+    async def csrf_token(self) -> str:
+        """Returns CSRF token."""
+        if self._csrf_token is None:
+            await self._acquire_tokens_for_forgery()
+        return self._csrf_token
+
+    @property
+    async def forever_alone_session(self) -> str:
+        """Returns some sort of token related to CSRF, no idea myself."""
+        if self._forever_alone_session is None:
+            await self._acquire_tokens_for_forgery()
+        return self._forever_alone_session
+
+    @property
+    async def forged_cookies(self) -> dict:
+        """Forged cookies for making API calls."""
+        return {
+            "XSRF-TOKEN": await self.xsrf_token,
+            "forever_alone_session": await self.forever_alone_session
+        }
+
+    @property
+    async def forged_headers(self) -> dict:
+        """Forged headers for making API calls."""
+        return {
+            "x-requested-with": "XMLHttpRequest",
+            "Referer": "https://mywaifulist.moe/dash",
+            "x-csrf-token": await self.csrf_token,
+            "x-xsrf-token": urllib.parse.unquote(await self.xsrf_token)
+        }
+
+    async def get_headers(self, json_as_string: bool = False) -> dict:
+        """Returns a modified copy of forged headers.
+        Mostly just an abstraction for dealing with crap POST request parameters."""
+        headers = (await self.forged_headers).copy()
+        if json_as_string:
+            headers.setdefault("Content-Type", "application/json")
+        return headers
+
+    async def get(self, endpoint: str) -> Response:
+        """Sends a GET request to Mywaifulist API endpoint."""
+        r = await self.client.get(
+            API_URL + endpoint,
+            headers=await self.forged_headers, cookies=await self.forged_cookies
+        )
+        return r
+
+    async def post(self, endpoint: str, json: dict) -> Response:
+        """Sends a POST request to Mywaifulist API endpoint."""
+        r = await self.client.post(
+            API_URL + endpoint, json=json,
+            headers=await self.forged_headers, cookies=await self.forged_cookies
+        )
+        return r
+
+    async def fetch_random_waifu(self) -> Waifu:
+        """Returns a random waifu."""
+        r = await self.client.get(f"{BASE_URL}/random", headers=await self.forged_headers, follow_redirects=True)
+        tree = etree.parse(StringIO(r.text), PARSER)
+        waifu_id = tree.xpath("//waifu-core")[0].attrib[':waifu-id']
+        waifu = await self.fetch_waifu_by_id(waifu_id)
+        self.cache[waifu_id] = waifu
+        return waifu
+
+    async def fetch_waifu_by_id(self, id: int) -> Waifu:
+        """Returns a waifu by the specified ID."""
+        if self.cache.get(id):
+            return self.cache.get(id)
+        else:
```

Comment:
Else can be commited since condition being true returns.

Reply:
i just think it looks neater but youre the boss here